resources/assets/coffee/react/beatmap-discussions/main.coffee

```diff
@@ -276,16 +276,17 @@ class BeatmapDiscussions.Main extends React.PureComponent
 
 
   discussions: =>
-    @cache.discussions ?= _.keyBy @state.beatmapset.discussions, 'id'
+    @cache.discussions ?= _ @state.beatmapset.discussions
+                            .filter (d) -> !_.isEmpty(d)
```

Comment:
Or `reject ... -> _.isEmpty ...`. And looks like `currentDiscussions` can use this function as well (and remove similar check in it).